pulser-core/pulser/register/special_layouts.py

```diff
@@ -84,16 +94,50 @@ def rectangular_register(
         if rows > self._rows or columns > self._columns:
             raise ValueError(
                 f"A '{rows}x{columns}' array doesn't fit a "
-                f"{self._rows}x{self._columns} SquareLatticeLayout."
+                f"{self._rows}x{self._columns} RectangularLatticeLayout."
             )
-        points = patterns.square_rect(rows, columns) * self._spacing
+        points = patterns.square_rect(rows, columns)
+        points[:, 0] = points[:, 0] * self._col_spacing
+        points[:, 1] = points[:, 1] * self._row_spacing
         trap_ids = self.get_traps_from_coordinates(*points)
         qubit_ids = [f"{prefix}{i}" for i in range(len(trap_ids))]
         return cast(
             pulser.Register,
             self.define_register(*trap_ids, qubit_ids=qubit_ids),
         )
 
+    def _to_dict(self) -> dict[str, Any]:
+        return obj_to_dict(
+            self, self._rows, self._columns, self._col_spacing,
+            self._row_spacing
+        )
```

Comment:
You'll have to add this new class here now: https://github.com/pasqal-io/Pulser/blob/71b8e0321371d0cfd311d90e5972292cc33977eb/pulser-core/pulser/json/supported.py#L70